File: export_sororities.py
#!/usr/bin/env python3
import os
from supabase import create_client
import csv

# Supabase credentials
SUPABASE_URL = "https://vvsawtexgpopqxgaqyxg.supabase.co"
SUPABASE_KEY = os.getenv("SUPABASE_KEY")

if not SUPABASE_KEY:
    print("ERROR: SUPABASE_KEY environment variable not set")
    print("Please set it with: export SUPABASE_KEY='your-key-here'")
    exit(1)

# Initialize Supabase client
supabase = create_client(SUPABASE_URL, SUPABASE_KEY)

print("Fetching sorority chapters from database...")

# Fetch all sorority chapters in batches
all_chapters = []
batch_size = 1000
offset = 0

while True:
    # Query chapters with sorority organization type
    response = supabase.table('chapters') \
        .select('universities(name), greek_organizations(name, organization_type), instagram_handle') \
        .eq('greek_organizations.organization_type', 'sorority') \
        .range(offset, offset + batch_size - 1) \
        .order('universities.name') \
        .execute()

    if not response.data:
        break

    all_chapters.extend(response.data)
    print(f"Fetched {len(all_chapters)} chapters so far...")

    if len(response.data) < batch_size:
        break

    offset += batch_size

print(f"\nTotal sorority chapters fetched: {len(all_chapters)}")

# Write to CSV
output_file = '/Users/jacksonfitzgerald/CollegeOrgNetwork/sorority_instagram_handles.csv'

with open(output_file, 'w', newline='', encoding='utf-8') as csvfile:
    writer = csv.writer(csvfile)
    writer.writerow(['College', 'Sorority Name', 'Instagram Handle'])

    for chapter in all_chapters:
        college_name = chapter.get('universities', {}).get('name', '') if chapter.get('universities') else ''
        sorority_name = chapter.get('greek_organizations', {}).get('name', '') if chapter.get('greek_organizations') else ''
        instagram_handle = chapter.get('instagram_handle', '') or ''

        writer.writerow([college_name, sorority_name, instagram_handle])

print(f"\n✅ CSV file created: {output_file}")
print(f"Total rows: {len(all_chapters)}")

File: backend/scripts/export_sororities.js
const { createClient } = require('@supabase/supabase-js');
const fs = require('fs');
const path = require('path');

require('dotenv').config({ path: path.join(__dirname, '../.env') });

const supabase = createClient(
  process.env.SUPABASE_URL,
  process.env.SUPABASE_SERVICE_ROLE_KEY || process.env.SUPABASE_ANON_KEY
);

async function exportSororities() {
  console.log('Fetching all sorority chapters...');

  const allChapters = [];
  const pageSize = 1000;
  let page = 0;
  let hasMore = true;

  while (hasMore) {
    const from = page * pageSize;
    const to = from + pageSize - 1;

    const { data, error } = await supabase
      .from('chapters')
      .select(`
        universities(name),
        greek_organizations(name, organization_type),
        instagram_handle
      `)
      .eq('greek_organizations.organization_type', 'sorority')
      .order('universities(name)')
      .range(from, to);

    if (error) {
      console.error('Error fetching data:', error);
      break;
    }

    if (!data || data.length === 0) {
      hasMore = false;
      break;
    }

    allChapters.push(...data);
    console.log(`Fetched ${allChapters.length} chapters so far...`);

    if (data.length < pageSize) {
      hasMore = false;
    }

    page++;
  }

  console.log(`\nTotal sorority chapters: ${allChapters.length}`);

  // Convert to CSV
  const csvRows = ['College,Sorority Name,Instagram Handle'];

  for (const chapter of allChapters) {
    const college = (chapter.universities?.name || '').replace(/"/g, '""');
    const sorority = (chapter.greek_organizations?.name || '').replace(/"/g, '""');
    const instagram = (chapter.instagram_handle || '').replace(/"/g, '""');

    csvRows.push(`"${college}","${sorority}","${instagram}"`);
  }

  const outputPath = path.join(__dirname, '../../sorority_instagram_handles.csv');
  fs.writeFileSync(outputPath, csvRows.join('\n'), 'utf-8');

  console.log(`\n✅ CSV file created: ${outputPath}`);
  console.log(`Total rows: ${allChapters.length}`);
}

exportSororities().catch(console.error);
